api/metrics_middleware.go

```diff
@@ -0,0 +1,58 @@
+package api
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+//From https://github.com/weaveworks/common/blob/81a1a4d158e60de72dbead600ec011fb90344f8c/middleware/instrument.go#L110-L136
+
+var handlerHistogram = promauto.NewHistogramVec(prometheus.HistogramOpts{
+	Name: "api_request_serve_time_seconds",
+	Help: "Time taken to serve HTTP request",
+}, []string{"method", "path", "status"})
+
+func metricMiddleware(next http.Handler) http.Handler {
```

Comment:
I'm not sure about this because maybe I'm missing something, but can't you use the official middleware here? https://godoc.org/github.com/prometheus/client_golang/prometheus/promhttp#InstrumentHandlerDuration